### PR TITLE
Add charm deploy button

### PR DESCRIPTION
**Pull request by huwshimi**
Added the "Added to canvas" button to charm details.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit `/:flags:/react/`
- open the mid-point
- click on a charm icon
- click "Add to canvas"
- a new service should be created

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3231/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3232/
Test PASSed.

**Comment by hatched** [comment]
:+1: QA :ok: 
When I add a service to the canvas from the charm details I get a number of the following warnings in the console: 

`Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op.`

File: jujugui/static/gui/src/app/components/entity-details/entity-details.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ YUI.add('entity-details', function() {
     fetchSuccess: function(models) {
       this.setState({waitingForFetch: false});
       if (models.length > 0) {
-        this.setState({entity: models[0].toEntity()});
+        var entity = models[0];
+        this.setState({entityModel: entity});
+        this.setState({entity: entity.toEntity()});
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/entity-details/entity-details.js]: I'd rather just store the model in state and then convert it to a POJO via the toEntity function in the `render` function. One source of truth.

File: jujugui/static/gui/src/app/views/utils.js [hunk 2]
```diff
@@ -2357,18 +2357,6 @@ YUI.add('juju-view-utils', function(Y) {
   };
 
   /**
-    Add a service to the canvas.
-
-    @method addService
-    @param {App} app The app object.
-    @param {String} charmId The id of the charm to add a service for.
-  */
-  utils.addService = function(app, charmId) {
```
Comment (kadams54) [inline jujugui/static/gui/src/app/views/utils.js]: Just to confirm, this isn't being used anywhere else in the codebase?

**Comment by kadams54** [comment]
:+1: with the comments above. I need this for code I'm working on, so landing this for now and I'll implement the changes suggested in a follow-up PR.

**Comment by kadams54** [comment]
:shipit: Thanks @huwshimi

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

File: jujugui/static/gui/src/app/views/utils.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/utils.js]: Nope, I addded it last week, we then stopped using it, so now I'm removing it.

**Pull request merged**